src/pocket/PocketECSCodePipeline.ts

```diff
@@ -256,33 +228,97 @@ export class PocketECSCodePipeline extends Resource {
         }
       ).json,
     });
+
+    return role;
   }
 
+  private getArtifactStore = () => [
+    {
+      location: this.pipelineArtifactBucket.bucket,
+      type: 'S3',
+      encryptionKey: { id: this.s3KmsAlias.arn, type: 'KMS' },
+    },
+  ];
+
   /**
-   * Creates a CodePipeline role.
+   * Get the source code from GitHub.
    * @private
    */
-  private createPipelineRole() {
-    return new iam.IamRole(this, 'codepipeline-role', {
-      name: `${this.config.prefix}-CodePipelineRole`,
-      assumeRolePolicy: new iam.DataAwsIamPolicyDocument(
-        this,
-        `codepipeline-assume-role-policy`,
-        {
-          statement: [
-            {
-              effect: 'Allow',
-              actions: ['sts:AssumeRole'],
-              principals: [
-                {
-                  identifiers: ['codepipeline.amazonaws.com'],
-                  type: 'Service',
-                },
-              ],
-            },
-          ],
-        }
-      ).json,
-    });
-  }
+  private getSourceStage = () => ({
+    name: 'Source',
+    action: [
+      {
+        name: 'GitHub_Checkout',
+        category: 'Source',
+        owner: 'AWS',
+        provider: 'CodeStarSourceConnection',
+        version: '1',
+        outputArtifacts: ['SourceOutput'],
+        configuration: {
+          ConnectionArn: this.config.source.codeStarConnectionArn,
+          FullRepositoryId: this.config.source.repository,
+          BranchName: this.config.source.branchName,
+          DetectChanges: 'false',
+        },
+        namespace: 'SourceVariables',
+      },
+    ],
+  });
+
+  /**
+   * Get a stage that deploys the infrastructure and ECS service.
+   * @private
+   */
+  private getDeployStage = (): codepipeline.CodepipelineStage => ({
+    name: 'Deploy',
+    action: [this.getDeployCdkAction(1), this.getDeployEcsAction(2)],
+  });
+
+  /**
+   * Get the CDK for Terraform deployment step that runs `terraform apply`.
+   * @private
+   */
+  private getDeployCdkAction = (
+    runOrder: number
+  ): codepipeline.CodepipelineStageAction => ({
+    name: 'Deploy_CDK',
+    category: 'Build',
+    owner: 'AWS',
+    provider: 'CodeBuild',
+    inputArtifacts: ['SourceOutput'],
+    outputArtifacts: ['CodeBuildOutput'],
+    version: '1',
+    configuration: {
+      ProjectName: this.codeBuildProjectName,
+      EnvironmentVariables: `[${JSON.stringify({
+        name: 'GIT_BRANCH',
+        value: '#{SourceVariables.BranchName}',
+      })}]`,
+    },
+    runOrder: runOrder,
+  });
+
+  /**
+   * Get the ECS CodeDeploy step that does a blue/green deployment.
+   * @private
+   */
+  private getDeployEcsAction = (
+    runOrder: number
```

Comment:
if these are only called once, and since everything else is hard-coded, would the code be simpler to hard-code the `runOrder` as well?